cmd/daytona/config/ssh_file.go

```diff
@@ -153,29 +153,32 @@ func EnsureSshConfigEntryAdded(profileId, workspaceName, projectName string, gpg
 		return err
 	}
 
-	sshDir := filepath.Join(sshHomeDir, ".ssh")
+	sshDir := filepath.Join(SshHomeDir, ".ssh")
 	configPath := filepath.Join(sshDir, "daytona_config")
 
 	knownHostsFile := getKnownHostsFile()
 
-	// Read existing content from the file
-	existingContent, err := os.ReadFile(configPath)
-	if err != nil && !os.IsNotExist(err) {
+	existingContent, err := ReadSshConfig(configPath)
+	if err != nil {
 		return err
 	}
 
-	// Generate SSH config entry without GPG forwarding
-	newContent, err := appendSshConfigEntry(configPath, profileId, workspaceName, projectName, knownHostsFile, false, string(existingContent))
-	if err != nil {
-		return err
+	regexWithoutGPG := regexp.MustCompile(fmt.Sprintf(`(?m)^Host %s-%s-%s\s*\n(?:\s+[^\n]*\n?)*`, profileId, workspaceName, projectName))
```

Comment:
![Screenshot 2024-12-05 at 11 39 18](https://github.com/user-attachments/assets/a919bac1-a5af-4321-8b60-4ab19a7bc94f)

From what I can see, you changed the way the ssh config entry is matched. This cause the issue from the screenshot.

The problem here is that `dtn` runs the binary with `go run` which creates a temp binary which is deleted after the run. This means I can only ssh into the workspace once in development mode.

We need to insure that the `ProxyCommand` line is always "regenerated" every time. Also, the user should not be able to edit that line. I suggest excluding it from the SSH config editor and then appending it before writing to the file.

Reply:
@Tpuljak This should work now!

Reply:
![Screenshot 2024-12-05 at 14 42 24](https://github.com/user-attachments/assets/5386c6aa-122d-4cbe-b127-ecd9b6a07fc8)

Let's also trim newlines that the user inputs. Then append the proxy command and one newline at the end.

Reply:
@Tpuljak you can check now!